administration/src/cards/CardBlueprint.ts

```diff
@@ -5,144 +5,205 @@ import { CardExtensions, CardInfo } from '../generated/card_pb'
+    }),
+    {}
+  )
+
+  return {
+    id: id ?? createRandomId(),
+    fullName: fullName ?? '',
+    extensions: {
+      ...defaultExtensions,
+      ...(extensions ?? {}),
+    },
+    expirationDate: expirationDate ?? defaultExpirationDate,
   }
+}
 
-  setValue(key: string, value: string): void {
-    switch (key) {
-      case this.cardConfig.nameColumnName:
-        this.fullName = value
-        break
-      case this.cardConfig.expiryColumnName:
-        this.setExpirationDate(value)
-        break
-      default:
-        const extensionIdx = this.cardConfig.extensionColumnNames.indexOf(key)
-        if (extensionIdx === -1) {
-          return
-        }
-        this.extensions[extensionIdx].fromString(value)
-    }
-  }
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const getExtensions = ({ extensions }: CardBlueprint) => {
+  const extensionKeys = Object.keys(extensions) as (keyof typeof Extensions)[]
+  return extensionKeys.map(key => {
+    const extensionObject = Extensions.find(extension => extension.name === key)!
+    type ExtensionStateType = InferExtensionStateType<typeof extensionObject>
+    return { ...extensionObject, state: extensions } as Extension<ExtensionStateType> & { state: ExtensionStateType }
+  })
+}
 
-  initialize(region: Region): void {
-    this.extensions.forEach(ext => {
-      if (ext instanceof RegionExtension) {
-        ext.setInitialState(region)
-      } else {
-        ext.setInitialState()
-      }
-    })
-  }
+export const hasInfiniteLifetime = (cardBlueprint: CardBlueprint): boolean =>
+  getExtensions(cardBlueprint).some(extension => extension.causesInfiniteLifetime(extension.state))
 
-  hasInfiniteLifetime(): boolean {
-    return this.extensions.some(ext => ext.causesInfiniteLifetime())
-  }
+export const isFullNameValid = ({ fullName }: CardBlueprint): boolean => {
+  const encodedName = new TextEncoder().encode(fullName)
+  return fullName.length > 0 && encodedName.length <= MAX_ENCODED_NAME_LENGTH && fullName.length <= MAX_NAME_LENGTH
+}
 
-  isFullNameValid(): boolean {
-    const encodedName = new TextEncoder().encode(this.fullName)
-    return (
-      this.fullName.length > 0 &&
-      encodedName.length <= MAX_ENCODED_NAME_LENGTH &&
-      this.fullName.length <= MAX_NAME_LENGTH
-    )
+export const isExpirationDateValid = (cardBlueprint: CardBlueprint, nullable = false): boolean => {
+  const today = PlainDate.fromLocalDate(new Date())
+  const expirationDate =
+    cardBlueprint.expirationDate !== null ? PlainDate.fromDaysSinceEpoch(cardBlueprint.expirationDate) : null
+  const startDay = cardBlueprint.extensions.startDay
+  const startDayDate = startDay !== undefined ? PlainDate.fromDaysSinceEpoch(startDay) : null
+
+  // TODO Why should this not be checked for self service? Why is this allowed to be nullable for csv/self service?
+  // TODO Adjust isValid
```

Comment:
See other question, why is this allowed to be nullable in some cases? Why shouldn't this be checked for self service cards?

Reply:
hm i think the name shouldn't be nullable and the extension depend on which extension i think.
For selfServiceCards we have no expiration day in frontend